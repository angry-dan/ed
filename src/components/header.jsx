import React from "react";
const ComponentHeader = ({title, body}) =>
  <header className="header"><h1 className="header__title">{title}</h1><p className="header__body">{body}</p></header>;

class ComponentHeaderForm extends React.Component {
  handleChange(t) {
    return (event) => {
      this.props.onChange({[t]: event.target.value});
    }
  }

  render () {
    return <div className="scale-transition scale-in">
      <input type="text" name="title" value={this.props.title} onChange={this.handleChange('title')} />
      <input type="text" name="body" value={this.props.body} onChange={this.handleChange('body')} />
    </div>;
  }
}

export default {
  'description': 'Some header',
  'component': ComponentHeader,
  'componentForm': ComponentHeaderForm,
  'defaultState': {title: "Hello, world", body: "Lorem ipsum dolor etc"}
};
