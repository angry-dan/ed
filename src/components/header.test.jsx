import React from 'react';
import {ComponentHeader, ComponentHeaderForm} from "./header";
import renderer from 'react-test-renderer';

test('component renders correctly', () => {
  const tree = renderer.create(<ComponentHeader title="Some title" body="Some other body" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('component form renders correctly', () => {
  const tree = renderer.create(<ComponentHeaderForm title="Some title" body="Some other body" />).toJSON();
  expect(tree).toMatchSnapshot();
})
