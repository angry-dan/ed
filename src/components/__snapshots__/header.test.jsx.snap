// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component form renders correctly 1`] = `
<div
  className="scale-transition scale-in"
>
  <input
    name="title"
    onChange={[Function]}
    type="text"
    value="Some title"
  />
  <input
    name="body"
    onChange={[Function]}
    type="text"
    value="Some other body"
  />
</div>
`;

exports[`component renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Some title
  </h1>
  <p
    className="header__body"
  >
    Some other body
  </p>
</header>
`;
